cdap-ui/app/cdap/components/Experiments/store/ExperimentDetailActionCreator.js

```diff
@@ -207,134 +155,134 @@ function getSplitsInExperiment(experimentId) {
           splits
         }
       });
+    }, (err) => {
+      setExperimentDetailError(`Failed to get splits in experiment '${experimentId}': ${err.response || err}`);
     });
 }
 
-function getModelStatus(experimentId, modelId) {
+function pollModelStatus(experimentId, modelId) {
   return myExperimentsApi
     .pollModelStatus({
       namespace: getCurrentNamespace(),
       experimentId,
       modelId
     })
     .subscribe(modelStatus => {
+      removeModelsWithError(modelId);
       experimentDetailsStore.dispatch({
         type: EXPERIMENTDETAILACTIONS.SET_MODEL_STATUS,
         payload: {
           modelId,
           modelStatus
         }
       });
+    }, () => {
+      addModelsWithError(modelId);
     });
 }
 
-function setActiveModel(activeModelId) {
-  let state = experimentDetailsStore.getState();
-  let {splitDetails} = state.models.find(model => model.id === activeModelId);
-  if (!splitDetails) {
-    getSplitsInExperiment(state.name);
+function getModelStatus(experimentId, modelId) {
+  addModelsLoading(modelId);
+  myExperimentsApi
+    .getModelStatus({
+      namespace: getCurrentNamespace(),
+      experimentId,
+      modelId
+    })
+    .subscribe(modelStatus => {
+      removeModelsWithError(modelId);
+      experimentDetailsStore.dispatch({
```

Comment:
nit: Can we couple these two together? This will render two renders for the underlying components. I feel this is one of the downsides of redux. The number of actions you emit will be directly proportional to the number of updates react has to perform on the component. Right now we don't have any reselect libs to check for updates.

Reply:
Also should we remove models from error when we fetch the model status for the first time? May be `removeModelsWithError` can check if the model is already in the `modelsInError` array and perform an action if it does. I understand this is to clear an existing error but we could avoid performing actions if its not required.

Reply:
Not exactly sure how we can couple these together, since they modify different values in the store, and we don't want one action to modify multiple values right? For your second comment, I have removed `removeModelsWithError` call from the success callback of `pollModelStatus`, since that's when we fetch the model status for the first time.

Reply:
My last commit moves the code for `removeModelsWithError` inside the `SET_MODEL_STATUS` action reducer, so only one action is dispatched. Can you check and let me know if it's what you want?